migtests/tests/pg-tests/pg-codependent-schemas/validate

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python3
+
+import yb
+
+def main():
+	yb.run_checks(migration_completed_checks)
+
+
+#=============================================================================
+
+EXPECTED_ROW_COUNT_PUBLIC = {
+	'audit2': 4,              
+	'foreign_test': 2,
+ 	'list_active': 2,  
+ 	'list_archived': 1,  
+ 	'list_part': 5,   
+	'primary_test': 3,
+	'tt':4,
+	'v1':2,
+	'mv1':2,
+}
+EXPECTED_ROW_COUNT_SCHEMA2 = {
+	'audit': 4,
+	'foreign_test': 3,
+	'list_others': 2, 
+	'primary_test': 2,
+	'v1': 2,
+}
+
+def check_validations_per_schema(tgt,schema_name):
+
+	EXPECTED_ROW_COUNT = EXPECTED_ROW_COUNT_PUBLIC if schema_name == 'public' else EXPECTED_ROW_COUNT_SCHEMA2
+
+	table_list = tgt.get_table_names(schema_name)
+	print("table_list:", table_list)
+	assert (len(table_list) == 7 and schema_name == 'public') or (len(table_list) == 4 and schema_name == 'schema2')
+
+	view_list = tgt.get_objects_of_type("VIEW", schema_name)
+	print("view_list:", view_list)
+	assert len(view_list) == 1
+
+	mview_list = tgt.get_objects_of_type("MVIEW", schema_name)
+	print("mview_list:", mview_list)
+	assert (len(mview_list) == 1 and schema_name == 'public') or (len(mview_list) == 0 and schema_name == 'schema2')
+
+	for table_name, row_count in EXPECTED_ROW_COUNT.items():
+		cnt = tgt.get_row_count(table_name,schema_name)
+		print(f"table_name: {schema_name}.{table_name}, row_count: {cnt}")
+		assert row_count == cnt
+	
+	fetched_triggers = tgt.fetch_all_triggers(schema_name)
+	print(f"fetched triggers list - {fetched_triggers}")
+	assert fetched_triggers == ["audit_trigger"]
+
+	fetched_procedures_functions = tgt.fetch_all_procedures(schema_name)
+	print(f"count of fetched procedures/functions - {len(fetched_procedures_functions)}")
+	chk_conditions = (len(fetched_procedures_functions) == 2 and schema_name == 'schema2') or (len(fetched_procedures_functions) == 1 and schema_name == 'public')
+	assert chk_conditions == True
+
+	QUERY_FOREIGN_KEY_TEST = f"insert into {schema_name}.foreign_test values (5,1,7);"
+	chk_err_returned = tgt.run_query_and_chk_error(QUERY_FOREIGN_KEY_TEST, 23503)
+	print(f"Checking Foreign key test on schema {schema_name} - {chk_err_returned}")
+	assert chk_err_returned == True
+
+
+def migration_completed_checks(tgt):
+
+	check_validations_per_schema(tgt, "public")
+	check_validations_per_schema(tgt, "schema2")
+	
+	cnt_partitions = tgt.execute_query("SELECT count(DISTINCT tableoid::regclass) FROM list_part;")
```

Comment:
Do not abbreviate `count` as `cnt`. Always try to have names that are easier to pronounce.

Let's avoid running ad-hoc queries using the `tgt`. Define a method in `tgt` to get a list of names of partitions of a given <schema, table>. And check the `len(partition_list)`.